### ISSUE TITLE
Prefetch page content via Ajax on link hover

### ISSUE THREAD

**Issue by connecteev** [nuxt/nuxt.js#6241]
### What problem does this feature solve?

Instantclick (http://instantclick.io) is a nifty little js library (< 3kb gzipped) that pre-fetches pages via Ajax when you hover on links, based on the user's intent to click on it. A great example of a site that does that is dev.to (example: go to https://dev.to/t/vue, open up the console and notice the ajax requests when you hover on any link)
Does Nuxt support this natively? I see some mentions of "prefetching" in the API documentation but I don't see it happening on my nuxt instance (I am on nuxt-edge). Is this functionality available in Nuxt? 
https://nuxtjs.org/api/configuration-router#prefetchlinks

Is it off by default, perhaps? If it's not available, this would be a really nifty thing to add to the core library, perhaps as a configuration option.
It would also be helpful to get more information on what "prefetch" in the API documentation is referring to, and how one can see it in action.

<div align="right"><sub><em>This feature request is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c9652">#c9652</a>)</em></sub></div>

**Comment by manniL** [comment]
Hey :wave: 	
Nuxt is prefetching the JS chunks corresponding to the page by default (when a link is in the viewport and the user has a good connection). it does not execute `asyncData` or `fetch` of that page.

I agree, the docs need a chapter about it.

More info in https://github.com/nuxt/nuxt.js/pull/4574

**Comment by connecteev** [comment]
Thanks for the links @manniL. So there are 2 threads here:

1. Re: understanding current functionality: I agree, the nuxt documentation assumes a lot and has no real examples....I still struggle with it every day. I hope that changes in 3.x to make the framework more approachable to "non power developers"
So Nuxt prefetches chunks of js for every link that is in the viewport, and this is by default, is that correct? Looking at this [hello world example](https://github.com/nuxt/nuxt.js/pull/4574#issuecomment-449049047)
One thing I don't understand, why are scripts like /_nuxt/runtime.js and /_nuxt/vendors.app.js **both** preloaded and "deferred" on the same page?

2. I believe it still makes sense to keep this feature request open, and provide an option in Nuxt to turn on ```asyncData``` or ```fetch``` of the *entire* page, on certain actions like hover / hover with delay like www.instantclick.io does it? Would you agree?

Thanks..

**Comment by manniL** [comment]
> So Nuxt prefetches chunks of js for every link that is in the viewport, and this is by default, is that correct?

Yes but only if the connection is good (better than 2G) and no data saver is enabled. Also, the browser has to support IntersectionObserver.

>  I hope that changes in 3.x to make the framework more approachable to "non power developers"

What could we do to improve the docs in your opinion?

re 2: We can keep it open, sure.

**Comment by connecteev** [comment]
@manniL 
Re: improving the docs, I believe having dead-simple, real-world usage examples you can copy-paste from (ideally with mini-github repos and demos, maybe in codesandbox or an embedded fiddle) is going to be the # 1 thing to help people understand how to use different features in nuxt and also increase the usage of the framework. It's inevitably also going to reduce the number of support queries (like "how do i do this?" or "why does this not work?" that I find myself asking all the time) that the nuxt team has to deal with. Right now, I have no place to ask these questions (discord isn't very active, I feel awkward / guilty filing "issues" in github, and people don't always answer well, or at all, on stackoverflow...partly because, as amazing as nuxt is for SSR....few people use it because of the complexity...the barrier to entry is still so high). I've spent months in nuxt and watched paid courses and tutorials and I still don't know 90% of it.

Take a look at these tutorials on nuxt, which I learned a lot from:
https://dev.to/lilianaziolek/understanding-nuxt-vue-hooks-and-lifecycle-part-1-48lc
https://dev.to/lilianaziolek/understanding-nuxt-amp-vue-hooks-and-lifecycle-part-2-mmc
However, these 2 articles (and accompanying github repos), while a step in the right direction, are **still** a bit too technical for the level I am talking about. 

Imagine repurposing the entire documentation around real-world usage scenarios that are definitely going to get every developer excited. I know we get a bad rap for being "lazy", and maybe that's true to some degree, but I would argue that we have a burning desire for time and efficiency - we want to create the best possible experience for users, move quickly, and create maintainable code that stands the test of time and doesn't have to be re-written every 6-18 months (10 yrs ago my threshold would have been 3 yrs for a rewrite, things obviously change and move a lot quicker now). The converse of this is "being stuck" for hours, days, or worse, weeks, like I have experienced over and over - this is not shade on nuxt alone, but nuxt certainly requires "significantly more skill" than the average dev (like me) possesses. Some examples of real-world usage scenarios? These are some I have struggled with myself:
1. CRUD with APIs talking to a server / database (any back-end API/webservice would be fine for demo purposes, in my case I use laravel). CRUD is the bread and butter of web development.
2. end-to-end auth with a database (I am aware of auth.nuxtjs.org, but how do I get that working with a DB? What happens on the client side? the nuxt-server side? The API/webservice side? How to manage sessions, cookies, local storage - lots of fogginess here.. Again, this is the bread-and-butter of any website / app you may want to build. 
3. real-world examples of plugins
4. real-world examples of middleware
5. real-world examples of mixins
6. real-world examples of the store (for example, nuxtServerInit can be used to get the server time, or to retrieve data that will be reused throughout the app 
7. and a nice to have would be end-to-end integration examples with Vue front-end frameworks, like element, vuetify, iview, ant-design, etc
I don't want to overwhelm to the point where this becomes a TLDR, but hopefully, you see where I'm coming from. All said and done, you guys on the Nuxt team are doing an amazing job and I am appreciative...I would just appreciate the framework a lot more and become a fan for life if the above became a reality.

**Comment by tr1s** [comment]
Coming from Gatsby, I decided to make a project in Nuxt, and was suprised to see all the components pre-fetched! I like the concept of only prefetching on hover and it would be a nice feature to have in Nuxt!

**Comment by manniL** [comment]
> Coming from Gatsby, I decided to make a project in Nuxt, and was suprised to see all the components pre-fetched! I like the concept of only prefetching on hover and it would be a nice feature to have in Nuxt!

At the moment, prefetching works via intersection observer (when a `<nuxt-link>` is in the viewport).
But that's not related to the actual issue here 🙈

**Comment by simplenotezy** [comment]
This would be a cool feature!

**Comment by fahmifitu** [comment]
This is still a nice feature to have in 2022

**Comment by danielroe** [comment]
let's track in https://github.com/nuxt/nuxt/issues/27003.
